heir_py/mlir_emitter.py

```diff
@@ -0,0 +1,199 @@
+"""Emitter from numba IR (SSA) to MLIR."""
+
+import operator
+import textwrap
+
+from numba.core import ir
+
+
+class TextualMlirEmitter:
+    def __init__(self, ssa_ir):
+        self.ssa_ir = ssa_ir
+        self.temp_var_id = 0
+        self.numba_names_to_ssa_var_names = {}
+        self.globals_map = {}
+
+    def emit(self):
+        func_name = self.ssa_ir.func_id.func_name
+        # probably should use unique name...
+        # func_name = ssa_ir.func_id.unique_name
+
+        # FIXME: use inferred or explicit types for args
```

Comment:
I know the whole feature is WIP, but these should probably still have an issue number for tracking (maybe just a single "fix up python frontend issues" issue for now?)

Reply:
Done in c2b7faa9